mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td

```diff
@@ -188,11 +189,15 @@ def SYCL_GroupType : SYCL_Type<"Group", "group"> {
   let assemblyFormat = "`<` `[` $dimension `]` `,` `(` $body `)` `>`";
 }
 
+def VectorEltTy : AnyTypeOf<[I1, I8, I16, I32, I64, F16, F32, F64]>;
+def VectorSplatArg : MemRefOf<[VectorEltTy]>;
+
 def SYCL_VecType : SYCL_Type<"Vec", "vec"> {
   let parameters = (ins "::mlir::Type":$dataType,
                         "int":$numElements,
                         "llvm::SmallVector<mlir::Type, 4>":$body);
   let assemblyFormat = "`<` `[` $dataType `,` $numElements `]` `,` `(` $body `)` `>`";
+  let genVerifyDecl = 1;
```

Comment:
What's the difference between genVerifyDecl and hasVerifier?

Reply:
[From the MLIR documentation](https://mlir.llvm.org/docs/AttributesAndTypes/#verification):
```quote
If the genVerifyDecl field is set, additional verification methods are generated on the class.
```
I guess this is the equivalent for types.

Reply:
We cannot: `hasVerifier` is to be used with operations and `genVerifyDecl`, with types. We cannot be consistent here...

Reply:
oic, thanks for the clarification.